apm-agent-plugins/apm-vertx/apm-vertx3-plugin/src/main/java/co/elastic/apm/agent/vertx/v3/web/http1/HttpServerRequestImplEndInstrumentation.java

```diff
@@ -42,7 +42,7 @@
 
     @Override
     public ElementMatcher<? super MethodDescription> getMethodMatcher() {
-        return named("doEnd").and(takesNoArguments());
+        return named("doEnd").or(named("onEnd")).and(takesNoArguments());
```

Comment:
did we miss this initially? `onEnd` is available in 3.9 but not in 3.6.
How did you notice this?
Maybe we need a unit test for making sure context gets cleaned up properly, WDYT?

Reply:
This is exactly what this map migration does- it triggers our existing global recycling test assertions, thus uncovering such cases.
What we had before is a map that holds references to spans without counting these references, so the Vert.x API could change, causing spans to remain in the map and still get recycled. Now that we count these map references, if you don't properly remove them from the map, the recycling assertion causes the v3-latest tests to fail, which is how I found that.